pkg/domain/domain.go

```diff
@@ -1887,11 +1903,16 @@ func (do *Domain) LoadPrivilegeLoop(sctx sessionctx.Context) error {
 
 		var count int
 		for {
+			var userList []string
+			var resp clientv3.WatchResponse
 			ok := true
 			select {
 			case <-do.exit:
 				return
-			case _, ok = <-watchCh:
+			case resp, ok = <-watchCh:
+				if ok {
+					userList = decodePrivilegeEvent(resp)
+				}
 			case <-time.After(duration):
 			}
```

Comment:
line 1920: when domain closes, the watch goroutine will leak. So I prefer to use domain's context in `Watch`. And if we do this, re-watch should checking both `ok` and `do.ctx.Err()`

Reply:
done